src/core/markdown.js

```diff
@@ -200,6 +200,11 @@ function normalizePadding(text) {
         }
         return chop;
       }, chop);
+    for (const pre of doc.querySelectorAll("pre")) {
+      // HTML parser implicitly removes a newline after <pre> which breaks reindentation
+      // algorithm.
+      pre.prepend("\n");
```

Comment:
I wonder if we should only do this if there is there is no already a new line there, otherwise we might end up with too many new lines if one already exists.